test/file.ts

```diff
@@ -2532,79 +2530,236 @@ describe('File', () => {
+
+      beforeEach(() => {
+        CONFIG.version = 'v4';
+        file.getDate = () => NOW;
       });
-    });
 
-    it('should set correct settings if resumable', done => {
-      const config = Object.assign({}, CONFIG, {
-        action: 'resumable',
+      it('should create a v4 signed url when specified', done => {
+        const SCOPE = '20190318/auto/storage/goog4_request';
+        const CREDENTIAL = `${CLIENT_EMAIL}/${SCOPE}`;
+        const EXPECTED_QUERY_PARAM = [
+          'X-Goog-Algorithm=GOOG4-RSA-SHA256',
+          `X-Goog-Credential=${encodeURIComponent(CREDENTIAL)}`,
+          'X-Goog-Date=20190318T000000Z',
+          'X-Goog-Expires=2',
+          'X-Goog-SignedHeaders=host',
+        ].join('&');
+
+        const EXPECTED_CANONICAL_HEADERS = 'host:storage.googleapis.com\n';
+        const EXPECTED_SIGNED_HEADERS = 'host';
+
+        const CANONICAL_REQUEST = [
+          'GET',
+          `/${BUCKET.name}/${encodeURIComponent(file.name)}`,
+          EXPECTED_QUERY_PARAM,
+          EXPECTED_CANONICAL_HEADERS,
+          EXPECTED_SIGNED_HEADERS,
+          'UNSIGNED-PAYLOAD',
+        ].join('\n');
+
+        BUCKET.storage.authClient.sign = (blobToSign: string) => {
+          assert.deepStrictEqual(blobToSign, [
+            'GOOG4-RSA-SHA256',
+            '20190318T000000Z',
+            SCOPE,
+            crypto.createHash('sha256').update(CANONICAL_REQUEST).digest('hex'),
+          ].join('\n'));
+          return Promise.resolve('signature');
+        };
+
+        const config = Object.assign({}, CONFIG, {
+          expires: NOW.valueOf() + 2000,
+        });
+
+        file.getSignedUrl(config, (err: Error, signedUrl: string) => {
+          assert.ifError(err);
+          assert.strictEqual(typeof signedUrl, 'string');
+          done();
+        });
       });
 
-      BUCKET.storage.authClient.sign = (blobToSign: string) => {
-        assert.strictEqual(blobToSign.indexOf('POST'), 0);
-        assert(blobToSign.indexOf('x-goog-resumable:start') > -1);
-        done();
-      };
+      it('should fail for expirations beyond 7 days', () => {
+        const config = Object.assign({}, CONFIG, {
+          expires: NOW.valueOf() + 7.1 * 24 * 60 * 60 * 1000,
+        });
+        assert.throws(
+            () => {
+              file.getSignedUrl(config, () => {});
+            },
+            /Max allowed expiration is seven days/,
+        );
+      });
 
-      file.getSignedUrl(config, assert.ifError);
-    });
+      it('should add response-content-type parameter', done => {
+        const type = 'application/json';
 
-    it('should return an error if signBlob errors', done => {
-      const error = new Error('Error.');
+        const config = Object.assign({}, CONFIG, {
+          responseType: type,
+        });
 
-      BUCKET.storage.authClient.sign = () => {
-        return Promise.reject(error);
-      };
+        directoryFile.getSignedUrl(config, (_: Error, signedUrl: string) => {
```

Comment:
Why do you need the types on the callback?  If the types on the method are good, you should be able to infer this.

Reply:
`let directoryFile: any`, probably for easier monkey-patching methods :(

We should really refactor these "stubs" with `sinon`

Reply:
I think we agreed it's better not to try to refactor too much in a feature PR - fixing this with `sinon` would be quite a big change given the size of this test file.